File: stepik-union/src/main/java/org/stepik/core/courseFormat/StudyStatus.java
package org.stepik.core.courseFormat;

import org.jetbrains.annotations.Nullable;

public enum StudyStatus {
    UNCHECKED, SOLVED, FAILED, NEED_CHECK;

    public static StudyStatus of(@Nullable String status) {
        if (status == null) {
            return NEED_CHECK;
        }

        status = status.toLowerCase();

        if (status.equals("correct") || status.equals("solved")) {
            return SOLVED;
        }

        if (status.equals("wrong") || status.equals("failed")) {
            return FAILED;
        }

        return UNCHECKED;
    }
}

File: stepik-union/src/main/kotlin/org/stepik/core/courseFormat/StudyStatus.kt
package org.stepik.core.courseFormat

enum class StudyStatus {
    UNCHECKED, SOLVED, FAILED, NEED_CHECK;


    companion object {

        fun of(status: String?): StudyStatus {
            val myStatus: String = status ?: return NEED_CHECK

            return when (myStatus.toLowerCase()) {
                "correct", "solved" -> SOLVED
                "wrong", "failed" -> FAILED
                else -> UNCHECKED
            }
        }
    }
}
